marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java

```diff
@@ -400,8 +402,18 @@ public RunFlowResponse run(String jobId, String[] items, Map<String, Object> opt
                 }
             }
             catch(Exception e) {
-                e.printStackTrace();
-                throw new RuntimeException(e);
+                if(StringUtils.containsIgnoreCase(e.getMessage(), "Plugin error")){
+                    try {
+                        resp = objectMapper.readValue(((MarkLogicServerException)e).getFailedRequest().getMessage(), RunFlowResponse.class);
+                    } catch (IOException ex) {
+                        ex.printStackTrace();
+                        throw new RuntimeException(ex);
```

Comment:
I think that if an IO error occurs while trying to parse the contents of the original exception, we should throw the original exception, not the IO error. Otherwise we'll lose the original error. Could include a message in the wrapped RuntimeException - i.e. "Unexpected IO error while parsing exception from running flow; original exception: " + e.getMessage().

Reply:
Wrapped the original exception in RunTimeException and thrown that